### PR TITLE
binary search script

### PR DESCRIPTION
**Pull request by jean-philippe-martin**
This script will look for a small input that trips BaseRecalibrator. However, it can be adapted for debugging pretty much anything else, so long as you have two versions of the code: a "known good" one to compare against, and a "under test" one that has the bug you're trying to generate a minimal input for.

### CODE REVIEW

File: scripts/binary_search.sh [hunk 1]
```diff
@@ -0,0 +1,189 @@
+#!/bin/sh
+
+
+# 
+# This script does a binary search to look for the smallest input that will
+# break something.
+#
+# JPM
```
Comment (davidadamsphd) [inline scripts/binary_search.sh]: Remove developers names.

**Comment by jean-philippe-martin** [comment]
Unless I hear otherwise, I'll merge when tests turn green.

**Comment by akiezun** [comment]
go for it. I wonder if any of the delta debugging tools would have come handy here: https://en.wikipedia.org/wiki/Delta_Debugging

**Comment by davidadamsphd** [comment]
LGTM (sorry I was out sick yesterday).
Nice to haves before submitting:
 - more commenting throughout the code (I find bash opaque) 
 - maybe make some of the files program args (I'm not so sure since this is going to be so run-to-run specific)

**Comment by jean-philippe-martin** [comment]
Ah, well, if bash is opaque to you then that's not the right file for you! This script has to be adapted before it can be useful, generally, unless I guess you're looking for the exact same bug I was looking for at the time.

I'll try to add more comments.

**Comment by akiezun** [comment]
can we migrate this to hellbender-dataflow?

**Comment by jean-philippe-martin** [comment]
@akiezun, the only thing Dataflow-specific with the script is that the currently-filled-in code is to look for a bug in the Dataflow code. But really the idea is that you'd change that bit to have it call whatever you're working on at the time. Are you sure you want me to move the file?

**Comment by akiezun** [comment]
no, you can merge as is if you want

**Comment by jean-philippe-martin** [comment]
tests are green, merging.

**Pull request merged**